hoomd/mpcd/ParticleData.cc

```diff
@@ -468,24 +468,36 @@ void mpcd::ParticleData::takeSnapshot(std::shared_ptr<mpcd::ParticleDataSnapshot
             // allocate memory in snapshot
             snapshot->resize(getNGlobal());
 
-            // write back into the snapshot in tag order, don't really care about cache coherency
+            // sorting particle rank and indices by it's tag
+            std::vector<std::tuple<unsigned int, unsigned int, unsigned int>> sorted_tags(getNGlobal());
+            unsigned int total_idx = 0;
+
             for (unsigned int rank_idx = 0; rank_idx < n_ranks; ++rank_idx)
                 {
                 const unsigned int N = pos_proc[rank_idx].size();
                 for (unsigned int idx = 0; idx < N; ++idx)
                     {
-                    const unsigned int snap_idx = tag_proc[rank_idx][idx];
+                    sorted_tags[total_idx] = std::make_tuple(tag_proc[rank_idx][idx], rank_idx, idx);
+                    total_idx++;
+                    }
+                }
+            std::sort(sorted_tags.begin(), sorted_tags.end());
 
-                    // make sure the position stored in the snapshot is within the boundaries
-                    Scalar3 pos_i = pos_proc[rank_idx][idx];
-                    int3 img = make_int3(0,0,0);
-                    global_box.wrap(pos_i,img);
+            // write back into the snapshot in the index order of that tag, don't really care about cache coherency
+            for (unsigned int snap_idx = 0; snap_idx < getNGlobal() ; ++snap_idx)
+                {
+                const unsigned int idx = std::get<2>(sorted_tags[snap_idx]);
+                const unsigned int rank_idx = std::get<1>(sorted_tags[snap_idx]);
```

Comment:
This makes more sense to read in the order they are stashed
```suggestion
                const unsigned int rank_idx = std::get<1>(sorted_tags[snap_idx]);
                const unsigned int idx = std::get<2>(sorted_tags[snap_idx]);
```